Sources/SwiftDocCTestUtilities/XCTestCase+TemporaryDirectory.swift

```diff
@@ -0,0 +1,62 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+
+// These helpers methods exist to put temp files for different test executions in different locations when running in Swift CI.
+
+public extension XCTestCase {
+    
+    @available(*, deprecated, message: "Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.")
+    func NSTemporaryDirectory() -> String {
+        return Foundation.NSTemporaryDirectory()
+    }
+    
+    /// Creates a new temporary directory and returns the URL of that directory.
+    ///
+    /// At the end of the test the temporary directory is automatically removed.
```

Comment:
I guess this means we should only use `createTemporaryDirectory` within test methods and it should not be called in static contexts or instance properties of an `XCTestCase`. I think we should clarify that in the documentation

Reply:
That's a good idea. These are for a special purpose.

Reply:
Coming back to this. I think it should be safe to call this wherever it is accessible.

Reply:
I think you're right. It seems that properties of test classes get re-initialized every test run. Neat!